rotkehlchen/chain/evm/transactions.py

```diff
@@ -541,6 +550,53 @@ def ensure_tx_data_exists(
         tx_receipt = self.dbevmtx.get_receipt(cursor, tx_hash, self.evm_inquirer.chain_id)
         return tx_data, tx_receipt  # type: ignore  # tx_data can't be None here
 
+    def get_and_ensure_internal_txns_of_parent_in_db(
+            self,
+            tx_hash: 'EVMTxHash',
+            to_address: 'ChecksumEvmAddress',
+            for_address: 'ChecksumEvmAddress',
+    ) -> list[EvmInternalTransaction]:
+        """Queries the internal transactions for an address of a parent hash where receiver is
+        to_address, saves them in the DB and returns them.
+
+        May raise:
+        - RemoteError if there is a problem querying the data sources or transaction hash does
+        not exist."""
+        db_internal_txs = self.dbevmtx.get_evm_internal_transactions(
+            parent_tx_hash=tx_hash,
+            blockchain=CHAINID_TO_SUPPORTED_BLOCKCHAIN[self.evm_inquirer.chain_id],
+        )
+        if len(db_internal_txs) > 0:
+            return db_internal_txs
+
+        # check cache if this parent hash was queried before for this receiver and affected address
+        with self.database.conn.read_ctx() as cursor:
+            affected_address = self.database.get_dynamic_cache(
+                cursor=cursor,
+                name=DBCacheDynamic.EXTRA_INTERNAL_TX,
+                tx_hash=tx_hash,
+                receiver=to_address,
+            )
+        if affected_address == for_address:  # if we have queried them before
+            return db_internal_txs
```

Comment:
for clairity this means that there was none in the db and there aren't any new so

```suggestion
            return []
```